nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -0,0 +1,305 @@
+
+            if (getSecretValueResult.getSecretString() == null) {
+                logger.debug("Secret [{}] is not configured", secretName);
+                return groups;
+            }
+
+            final ObjectNode secretObject = parseSecret(getSecretValueResult.getSecretString());
+            if (secretObject == null) {
+                logger.debug("Secret [{}] is not in the expected JSON key/value format", secretName);
+                return groups;
+            }
+
+            for (final Iterator<Map.Entry<String, JsonNode>> it = secretObject.fields(); it.hasNext(); ) {
+                final Map.Entry<String, JsonNode> field = it.next();
+                final String parameterName = field.getKey();
+                final String parameterValue = field.getValue().textValue();
+                if (parameterValue == null) {
+                    logger.debug("Secret [{}] key [{}] has no value", secretName, parameterValue);
+                    continue;
+                }
+
+                parameters.add(createParameter(parameterName, parameterValue));
+            }
+
+            groups.add(new ParameterGroup(secretName, parameters));
+
+            return groups;
+        } catch (final ResourceNotFoundException e) {
+            logger.error("Secret [{}] not found", secretName);
+            throw new IllegalStateException(String.format("Secret %s not found", secretName), e);
+        } catch (final AWSSecretsManagerException e) {
+            logger.error("Error retrieving secret [{}]", secretName);
+            throw new IllegalStateException("Error retrieving secret " + secretName, e);
+        }
+    }
+
+    private Parameter createParameter(final String parameterName, final String parameterValue) {
+        final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder().name(parameterName).build();
+        return new Parameter(parameterDescriptor, parameterValue, null, true);
+    }
+
+    protected ClientConfiguration createConfiguration(final ConfigurationContext context) {
+        final ClientConfiguration config = new ClientConfiguration();
+        config.setMaxErrorRetry(0);
+        config.setUserAgentPrefix(DEFAULT_USER_AGENT);
+        config.setProtocol(DEFAULT_PROTOCOL);
+        final int commsTimeout = context.getProperty(TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        config.setConnectionTimeout(commsTimeout);
+        config.setSocketTimeout(commsTimeout);
+
+        if (this.getSupportedPropertyDescriptors().contains(SSL_CONTEXT_SERVICE)) {
+            final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+            if (sslContextService != null) {
+                final SSLContext sslContext = sslContextService.createContext();
+                SdkTLSSocketFactory sdkTLSSocketFactory = new SdkTLSSocketFactory(sslContext, new DefaultHostnameVerifier());
```

Comment:
Instead of instantiating a new `DefaultHostnameVerifier`, it looks one of the static properties on `SdkTLSSocketFactory` could be used. The Browser Compatible version seems like the best option.